schema/types/isolator_linux_specific.go

```diff
@@ -149,15 +174,148 @@ func NewLinuxCapabilitiesRevokeSet(caps ...string) (*LinuxCapabilitiesRevokeSet,
 	return &l, nil
 }
 
-func (l LinuxCapabilitiesRevokeSet) AsIsolator() Isolator {
+func (l LinuxCapabilitiesRevokeSet) AsIsolator() (*Isolator, error) {
 	b, err := json.Marshal(l.linuxCapabilitiesSetBase.val)
 	if err != nil {
-		panic(err)
+		return nil, err
 	}
 	rm := json.RawMessage(b)
-	return Isolator{
+	return &Isolator{
 		Name:     LinuxCapabilitiesRevokeSetName,
 		ValueRaw: &rm,
 		value:    &l,
+	}, nil
+}
+
+type LinuxSeccompSet interface {
+	Set() []LinuxSeccompEntry
+	Errno() LinuxSeccompErrno
+	AssertValid() error
+}
+
+type LinuxSeccompEntry string
+type LinuxSeccompErrno string
+
+type linuxSeccompValue struct {
+	Set   []LinuxSeccompEntry `json:"set"`
+	Errno LinuxSeccompErrno   `json:"errno"`
+}
+
+type linuxSeccompBase struct {
+	val linuxSeccompValue
+}
+
+func (l linuxSeccompBase) MaxInstances() uint {
+	return 1
+}
+
+func (l linuxSeccompBase) AssertValid() error {
+	if len(l.val.Set) == 0 {
+		return errors.New("empty set")
+	}
+	if l.val.Errno == "" {
+		return nil
+	}
+	for _, c := range l.val.Errno {
+		if !unicode.IsUpper(c) {
```

Comment:
I'd rather return a more specific error - e.g "Errno must be an upper case string" or something